src/aklite_client_ext.cc

```diff
@@ -26,25 +27,25 @@
 // Returns the target that should be installed, if any.
 // It might be an updated version, a rollback target, or even the currently installed target, in case we need to sync
 // apps
-GetTargetToInstallResult AkliteClientExt::GetTargetToInstall(const LocalUpdateSource* local_update_source, int version,
+GetTargetToInstallResult AkliteClientExt::GetTargetToInstall(const CheckInResult& check_in_result, int version,
                                                              const std::string& target_name, bool allow_bad_target,
-                                                             bool force_apps_sync) {
-  // Tell CheckIn and CheckInLocal methods to not call notifyTufUpdateFinished in case of success
-  usingUpdateClientApi = true;
+                                                             bool force_apps_sync, bool is_offline_mode) {
+  std::string err;
+  if (!check_in_result) {
```

Comment:
It looks like redundant check since it will fail at line 44-48 anyway. Also, it seems like ` client_->notifyTufUpdateFinished(err);` won't be invoked in this case, while we want  to invoke this callback even if CheckIn fails.